lib/rugged/tag.rb

```diff
@@ -1,5 +1,24 @@
 module Rugged
   class Tag < Rugged::Reference
+    GPG_SIGNATURE_PREFIX = "-----BEGIN PGP SIGNATURE-----".freeze
+
+    def self.extract_signature(repo, oid, prefix=GPG_SIGNATURE_PREFIX)
+      object = repo.read(oid)
+
+      unless object.type == :tag
+        raise GitRPC::InvalidObject, "Invalid object type #{object.type}, expected tag"
+      end
+
+      unless index = object.data.index(prefix)
+        raise GitRPC::InvalidObject, "Tag does not contain signature"
```

Comment:
I think here you want to return `nil`, instead of raising. An exception is something exceptional. It would be more idiomatic to return nothing if there's nothing to return.